src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -260,7 +260,7 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
         public override void VisitDeclarationExpression(DeclarationExpressionSyntax node)
         {
             var argumentSyntax = (ArgumentSyntax)node?.Parent;
-            var argumentListSyntax = (BaseArgumentListSyntax)argumentSyntax?.Parent;
+            var argumentListSyntax = argumentSyntax.Parent as BaseArgumentListSyntax;
             var variable = MakeOutVariable(node, argumentListSyntax, _nodeToBind);
```

Comment:
>var variable = MakeOutVariable(node, argumentListSyntax, _nodeToBind); [](start = 12, length = 70)

The ```MakeOutVariable``` name is now misleading, please rename so that the name wouldn't make people believe that the function is used only for legitimate Out Vars.